### DISCUSSION TITLE
Upcoming v4 changes

### DISCUSSION THREAD

**Discussion by creazy231** [alexzaganelli/strapi-plugin-email-designer#55] [Ideas]
Since everyone should know by now - Strapi v4 is finally coming. Since email designer is such a nice and important plugin for so many of us we should definitely migrate asap.
I'd suggest that we create a list with todos to migrate email designer to the new v4 plugin system to be able to contribute with all of the repo contributors which wants to help :)

What do you think @alexzaganelli ?

**Comment by alexzaganelli** [comment]
Hi super @creazy231 !
I agree, it's very very important!! I'm currently abroad and I'm not able to do that but I could do as soon as I return home.
Do you already have some experience on that?

**Reply by creazy231** [reply]
Had no chance to take my hands on v4 right now but I'd do so when the public beta of v4 begins.
I have to migrate my own plugin (https://github.com/creazy231/strapi-plugin-raw-query) too so I'll dive into it and report anything I'll face here to create some kind of a todo list then.

**Reply by alexzaganelli** [reply]
Thank you so much @creazy231 !!

**Reply by meck93** [reply]
I would be happy to help as well 👍

**Comment by s3n-w6i** [comment]
You can now find the docs for v4 here: [https://docs-next.strapi.io](https://docs-next.strapi.io)

They look pretty incomplete to me at the moment, but this will surely get better over time.

**Comment by creazy231** [comment]
I took the time to rewrite my own plugin for v4.beta.4+, but since there is no official migration guide, I would wait until one is released.

Raw Query Beta v4: https://github.com/creazy231/strapi-plugin-raw-query/tree/beta-v4

**Comment by s3n-w6i** [comment]
https://strapi.io/blog/v4-plugin-migration-guide

**Comment by Flowko** [comment]
any news about this ? since strapi v4 is officially live now

**Reply by jnetRepo** [reply]
Hi  @alexzaganelli @creazy231
Any chance that this can be released for v4?

**Reply by alexzaganelli** [reply]
Hi guys, the new plugin for the v4 is coming. Please be patient

**Reply by akilshadi** [reply]
Hey @alexzaganelli, is there any date? i am developing a plugin, which depends on your plugin to design the email template

**Reply by alexzaganelli** [reply]
> Hey @alexzaganelli, is there any date? i am developing a plugin, which depends on your plugin to design the email template

I hope by the end of the next week. Unfortunately I don't have so much time to work on OSS currently...

**Comment by creazy231** [comment]
https://docs.strapi.io/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin-migration.html

Official Strapi V4 Migration Guide for Plugins 👌🏻

**Comment by alexzaganelli** [comment]
@creazy231 @meck93 @pr0gr8mm3r @Flowko @jnetRepo @akilshadi @GuilleAngulo @nihey  @nikzaugg etc

Hi guys, the migration took a lot of effort and many many nights but finally I should have reached a good point to start the testing.

Please follow the README to install and activate the plugin:
https://github.com/alexzaganelli/strapi-plugin-email-designer#-installation

Let me know if it's working from your side 👀

**Reply by creazy231** [reply]
Will migrate boltapi.com to Strapi v4 as soon as I have time for sideprojects and will take a look into it in the same step 👍 
Thanks for your hard work and the effort you put into this

**Reply by team-wealthian** [reply]
@alexzaganelli  I have tried to follow the steps mentioned above as well as setting in plugin.js mentioned in the home page...but still I am not able to view email-designer option in the plugins section. Please help

**Reply by alexzaganelli** [reply]
> @alexzaganelli  I have tried to follow the steps mentioned above as well as setting in plugin.js mentioned in the home page...but still I am not able to view email-designer option in the plugins section. Please help

Maybe you have to build with

`yarn build`

Or

`npm run build`

Anyway I've updated the README and you should be able to install and activate the plugin following the docs.

**Reply by nihey** [reply]
Thanks! It is good to know that this project is being constantly updated.

We will gradually migrate to Strapi V4 over time, and will make sure to update your plugin as well.

**Reply by nihey** [reply]
We've migrated to Strapi v4 and successfully send some emails using `strapi-plugin-email-designer`. Everything working great so far! Thanks for your effort in migrating it!

**Comment by akilshadi** [comment]
In Github it is said that templateReferenceId must be used, but two templates can have the same templateReferenceId, how can make different between them?

**Reply by Flowko** [reply]
@akilshadi use templateId, thats unique, u can get it from here
![image](https://user-images.githubusercontent.com/35883748/154088244-5c85b00c-fc67-465f-ae3a-f3a24674e2c5.png)
